internal/cmd/config/config.go

```diff
@@ -128,6 +129,19 @@ type Controller struct {
 	StatusGracePeriodDuration time.Duration `hcl:"-"`
 }
 
+func (c *Controller) InitNameIfEmpty() (string, error) {
+	if c == nil {
+		return "", fmt.Errorf("controller config is empty")
```

Comment:
It's not the config that's empty, it's the controller itself. More to the point though, can you just remove the specific controller/worker references and combine this into a shared function in the `base` package? Maybe:

`c.Name, err = base.InitNameIfEmpty(c.Name)`

Reply:
refactored the bits that generates into a shared func but I kept the paired down receiver functions.  I'm confused about it not being the Config that's empty since it's `c.Config.Controller` or `w.Config.Worker`

Reply:
The first line there -- `if c == nil` -- c is a `*Controller`. So the check is on that pointer, not `c.Config`. Does that make sense?